### ISSUE TITLE
No Battle.net gateways

### ISSUE THREAD

**Issue by iopq** [awesie/wine-starcraft#2]
The game runs fine on a new Wine prefix, but even when I use Windows XP with "Hide Wine version from applications" enabled I don't  have gateways and the Battle.net OK button is grayed out.

If I select LAN and then go back and click OK on Battle.net I get "The selected network provider could not be initialized."

Any idea how I can play online?

**Comment by demirk4n** [comment]
@iopq Can you try keeping "Default Settings" in "Windows 7" but setting "StarCraft.exe" to "Windows XP with "Hide Wine version" option checked?

Here are screenshots from my current winecfg.
http://imgur.com/a/qz1GN

**Comment by BadBrainJohnson** [comment]
@iopq Had the same issue with battle.net because of secur32 override. Just dropped it and now I can connect and login.

**Comment by iopq** [comment]
@demirk4n that's exactly my setup, that's not the issue I'm having

@BadBrainJohnson how do I do that?

**Comment by iopq** [comment]
OK, I found out the issue. There's actually a 30 second delay before each modal is populated. That means I have to wait for 30 seconds for gateways to show up. Then I have to wait 30 more seconds to log in. 30 more seconds for games to show up, etc.

**Comment by BadBrainJohnson** [comment]
@iopq oh:( I've been fiddling with it the whole day...I know that feeling. Sure there is no general networking issue? (run a different app in your wine-prefix to make sure, portforwarding shouldn't be an issue just to connect to battle.net) 

I can't exactly reproduce my steps to a working setup (on Ubuntu Zesty), but I'll try to put up a test with a guide on WineHQ in the next couple of days.

**Comment by iopq** [comment]
No, it doesn't affect the game, it just affects finding a game, it takes a long time since each modal has a delay

**Comment by rsanden** [comment]
Thank you Andrew for your work on this project. For what it's worth as a data point, I'm using the following approach and did not see any such delays: https://us.battle.net/forums/en/starcraft/topic/20754528299 .

**Comment by muesli4** [comment]
I only did create a new *wine prefix*, then everything worked out of the box with *Windows XP* and only a *default settings* entry (I didn't even check *Hide Wine version from applications* by accident). I've been playing for a few days without a single crash. Seems to be very stable.

**Comment by awesie** [comment]
I've seen the delays myself, not sure why they happen. Also, all of the patches necessary for Starcraft should now be in wine-staging master branch.

**Comment by iopq** [comment]
It loads the damn HTML over and over again (through localhost, but somehow it's slow anyway), and I don't see the gateways until the fonts load!

**Comment by w3chtml** [comment]
The same problem, loads html too slow - each menu/action up to few minutes. In console I see each time:
[0511/140604:VERBOSE1:resource_loader.cc(385)] OnResponseStarted: http://127.0.0.1:43867/gamedata/webui/dist/lib/fonts/Eurostile-Reg.otf
[0511/140604:VERBOSE1:buffered_resource_handler.cc(275)] To buffer: http://127.0.0.1:43867/gamedata/webui/dist/lib/fonts/Eurostile-Reg.otf
[0511/140604:VERBOSE1:resource_loader.cc(455)] OnReadCompleted: "http://127.0.0.1:43867/gamedata/webui/dist/lib/fonts/Eurostile-Reg.otf" bytes_read = 32768
[0511/140604:VERBOSE1:resource_loader.cc(455)] OnReadCompleted: "http://127.0.0.1:43867/gamedata/webui/dist/lib/fonts/Eurostile-Reg.otf" bytes_read = 32768
[0511/140604:VERBOSE1:resource_loader.cc(455)] OnReadCompleted: "http://127.0.0.1:43867/gamedata/webui/dist/lib/fonts/Eurostile-Reg.otf" bytes_read = 620
[0511/140604:VERBOSE1:resource_loader.cc(455)] OnReadCompleted: "http://127.0.0.1:43867/gamedata/webui/dist/lib/fonts/Eurostile-Reg.otf" bytes_read = 0
[0511/140604:VERBOSE1:resource_loader.cc(811)] ResponseCompleted: http://127.0.0.1:43867/gamedata/webui/dist/lib/fonts/Eurostile-Reg.otf

It seems BW search fonts, wait till timeout and use any font..
Me: Arch linux, runned `winetricks corefonts` without any progress.

**Comment by demirk4n** [comment]
@w3chtml have you tried the Aur package; https://aur.archlinux.org/packages/wine-starcraft-git/ ?

**Comment by w3chtml** [comment]
@demirk4n yes. My report with this package. Sorry not mention it.

**Comment by demirk4n** [comment]
@w3chtml Just a quick question, do you have the "wine-staging" installed from the official repo as well? If not, can you try to install that, too? I'm using the same git package side-by-side with the official one without any issues you mentioned.

**Comment by w3chtml** [comment]
@demirk4n It seems like this package in conflict with `wine`. Not right now I'll check this package

**Comment by demirk4n** [comment]
@awesie Hi Andrew,
Wine 2.8 is recently announced and in the change log, I saw your contributions. Does that include your starcraft related patch? If not, I guess we should wait for the wine-staging 2.8 release.

https://www.winehq.org/announce/2.8
Andrew Wesie (1):
      setupapi: Fix SetupDiGetDeviceRegistryProperty if property does not exist.

**Comment by awesie** [comment]
One of the simple changes was committed to Wine. The remaining patches are in Wine staging.

**Comment by w3chtml** [comment]
My final report with success:
Like was suggested @demirk4n installed `wine-staging`  - no changes
Then found https://www.linux.org.ru/forum/games/12855310 (Russian) where suggested install `lib32-libldap`, so `yaourt -S lib32-libldap` and success.

Fast load in Bnet, no delays at all! 
Hope it can save somebody time.
thanks all.

**Comment by jasonkhanlar** [comment]
```
fixme:winsock:WS_EnterSingleProtocolW unknown Protocol <0x00000000>
fixme:service:svcctl_EnumServicesStatusW resume index not supported
fixme:service:svcctl_EnumServicesStatusW resume index not supported
fixme:ntdll:server_ioctl_file Unsupported ioctl 48000022 (device=4800 access=0 func=8 method=2)
fixme:winsock:server_ioctl_sock Unsupported ioctl 48000022 (device=4800 access=0 func=8 method=2)
fixme:winsock:WSAIoctl unsupported WS_IOCTL cmd (_WSAIOR(IOC_WS2, 34))
fixme:winsock:WS_EnterSingleProtocolW unknown Protocol <0x00000000>
```

http://i.imgur.com/tSAtCk5.jpg

Even waiting 60 seconds for something to appear, nothing appears in the bottom-left modal.

When I start starcraft, a dialog appears before the game loads:
```
A StarCraft service has failed to launch. Your antivirus software may be interfering with a required service. Some functionality of StarCraft may be disabled.
```

In console the following appears at the same time as the above dialog:

```
fixme:advapi:ReportEventA (0xcafe4242,0x0004,0x0000,0x00000064,(nil),0x0001,0x00000000,0x79e548,(nil)): stub
fixme:advapi:ReportEventW (0xcafe4242,0x0004,0x0000,0x00000064,(nil),0x0001,0x00000000,0x113078,(nil)): stub
fixme:advapi:DeregisterEventSource (0xcafe4242) stub
```

**Comment by iopq** [comment]
> A StarCraft service has failed to launch. Your antivirus software may be interfering with a required service. Some functionality of StarCraft may be disabled.

This is what happens when the version is wrong.

When you set it to Windows XP you may or may not need to hide Wine version from application.

**Comment by jasonkhanlar** [comment]
Thanks!

Unchecking Hide Wine version from applications
and
Adding entry in Applications for StarCraft.exe using Windows XP

Battle.net now works!
